.github/workflows/upload_app_store.yml

```diff
@@ -0,0 +1,66 @@
+name: Upload IPA to App Store
+on: workflow_dispatch
+jobs:
+  check_branch:
+    name: "Check branch ('release*' or 'main')."
+    runs-on: macos-latest
+    steps:
+      - id: early_exit
+        if: ${{ github.ref != 'refs/heads/main' && startsWith(github.ref, 'refs/heads/release') != true }}
+        name: Early exit failure if not main or release branch.
+        run: exit 1
+  build_test_upload:
+    name: Build, test, and upload to App Store
+    needs: check_branch
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+            node-version: 16
+            cache: "npm"
+            cache-dependency-path: "**/package-lock.json"
+      - name: Build core and features
+        run: ./build.js
+        env:
+            POLYPOD_POLYPEDIA_REPORT_URL: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_URL }}
+            POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION }}
+      - name: Lint core and features
+        run: ./build.js lint
+      - name: Test core and features
+        run: xvfb-run --auto-servernum ./build.js test
+      - name: Lock xcode version
+        run: sudo xcode-select -switch /Applications/Xcode_12.4.app
+      - name: Run tests
+        working-directory: ios
+        run: |
+          set -o pipefail
```

Comment:
I didn't find a way to have `set -o pipefail` in Makefile, any suggestions are welcomed.

`set -o pipefail` is needed to properly handle xcodebuild failures, as it will terminate with valid exit code even when failed.